src/package/docker.ts

```diff
@@ -0,0 +1,201 @@
+  });
+  process.stderr?.on('data', (data) => {
+    logger.debug(data.toString());
+  });
+  await process;
+};
+
+const writePackageJSON = async (
+  logger: Logger,
+  outputDirectory: string,
+  message: string,
+  pluginVersion: string,
+  supportedTargets: SupportedTarget[],
+  plugin: Plugin,
+) => {
+  const packageJsonPath = path.join(outputDirectory, 'package.json');
+  logger.info(`Writing package.json to ${packageJsonPath}`);
+  await fs.writeFile(
+    packageJsonPath,
+    JSON.stringify(
+      {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        schema_version: 1,
+        name: plugin.name(),
+        team: plugin.team(),
+        kind: plugin.kind(),
+        version: pluginVersion,
+        message: message,
+        protocols: [3],
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        supported_targets: supportedTargets,
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        package_type: 'docker',
+      },
+      null,
+      2,
+    ),
+  );
+};
+
+const buildDockerfile = async (
+  logger: Logger,
+  outputDirectory: string,
+  pluginDirectory: string,
+  pluginVersion: string,
+  plugin: Plugin,
+) => {
+  const dockerfile = plugin.dockerFile();
+  const dockerFilePath = path.join(pluginDirectory, dockerfile);
+  if (!(await pathExists(dockerFilePath))) {
+    throw new Error(`docker file ${dockerFilePath} does not exist`);
+  }
+
+  logger.info(`Building ${plugin.buildTargets().length} targets`);
+  const supportedTargets = await pMap(
+    plugin.buildTargets(),
+    async ({ os, arch }) => {
+      const imageRepository = `registry.cloudquery.io/${plugin.team()}/${plugin.kind()}-${plugin.name()}`;
+      const imageTag = `${imageRepository}:${pluginVersion}-${os}-${arch}`;
+      const imageTar = `plugin-${plugin.name()}-${pluginVersion}-${os}-${arch}.tar`;
+      const imagePath = `${outputDirectory}/${imageTar}`;
+      logger.info(`Building docker image ${imageTag}`);
+      const dockerBuildArguments = [
+        'buildx',
+        'build',
+        '-t',
+        imageTag,
+        '--platform',
+        `${os}/${arch}`,
+        '-f',
+        dockerFilePath,
+        '.',
+        '--progress',
+        'plain',
+        '--load',
+      ];
+      logger.debug(`Running command 'docker ${dockerBuildArguments.join(' ')}'`);
+      await runDockerCommand(logger, dockerBuildArguments, pluginDirectory);
+      logger.debug(`Saving docker image ${imageTag} to ${imagePath}`);
+      const dockerSaveArguments = ['save', '-o', imagePath, imageTag];
+      logger.debug(`Running command 'docker ${dockerSaveArguments.join(' ')}'`);
+      await runDockerCommand(logger, dockerSaveArguments, pluginDirectory);
+      const checksum = await computeHash(imagePath);
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      return { os, arch, path: imageTar, checksum, docker_image_tag: imageTag };
```

Comment:
Added `docker_image_tag` as that is needed for the `push` command in the CLI.
We can also reconstruct the tag based on information we have in the CLI, but in case we ever change the tag structure I think this way is better